toolkit/tools/precacher/precacher.go

```diff
@@ -135,30 +135,30 @@ func downloadMissingPackages(rpmSnapshot *repocloner.RepoContents, packagesAvail
 	wg := new(sync.WaitGroup)
 	netOpsSemaphore := make(chan struct{}, concurrentNetOps)
 	results := make(chan downloadResult)
-	doneChannel := make(chan struct{})
```

Comment:
The `doneChannel` here is more of a synchronization primitive than a cancel signal. IMHO, you should leave this code as-is to avoid confusion.

That being said, you could simplify this code by calling `close(results)` instead of `close(done)`. Then in `monitorProgress` change the `result := <-results` to `result, ok := <-results` and use `ok` to break the loop.

Reply:
Agreed, always forget about the `ok` mechanism. Will use the results flow.